lib/ts/recipe/emailpassword/utils.ts

```diff
@@ -326,10 +326,10 @@ export function mergeFormFields(
         // If new field, push to mergeFormFields.
         if (isNewField) {
             mergedFormFields.push({
-                optional: false,
                 placeholder: userField.label,
                 validate: defaultValidate,
                 ...userField,
+                optional: Boolean(userField.optional),
```

Comment:
this change is not needed.. if you see, it's already being set to false by default.